src/syslog_mcp/tools/analysis/device_health_analyzer.py

```
@@ -0,0 +1,348 @@
+"""
+Device health analysis logic.
+
+This module provides pure business logic for analyzing device health data,
+including status assessment and health scoring.
+No data access or presentation logic - just analysis.
+"""
+
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, Optional
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)